src/submission_handlers/add-member.php

```diff
@@ -2,47 +2,67 @@
 include_once str_replace('/', DIRECTORY_SEPARATOR, 'includes/classes/file-utils.php');
 require_once FileUtils::normalizeFilePath('includes/classes/db-connector.php');
 require_once FileUtils::normalizeFilePath('includes/session-handler.php');
-require_once FileUtils::normalizeFilePath('includes/classes/session-manager.php');
 require_once FileUtils::normalizeFilePath('includes/classes/query-handler.php');
 require_once FileUtils::normalizeFilePath('includes/mailer.php');
 require_once FileUtils::normalizeFilePath('includes/classes/email-sender.php');
 
-    $conn = DatabaseConnection::connect();
+$conn = DatabaseConnection::connect();
+$emailError = '';
 
-    if ($_SERVER["REQUEST_METHOD"] == "POST") {
+if ($_SERVER["REQUEST_METHOD"] == "POST") {
+    if (isset($_POST["last_name"], $_POST["first_name"], $_POST["email"], $_POST["role"])) {
         $lastName = $_POST["last_name"];
         $firstName = $_POST["first_name"];
-        $middleName = $_POST["middle_name"];
+        $middleName = isset($_POST["middle_name"]) ? $_POST["middle_name"] : NULL;
+        $suffix = isset($_POST["suffix"]) ? $_POST["suffix"] : NULL;
         $email = $_POST["email"];
         $role = $_POST["role"];
-    
-      
-        $password = bin2hex(random_bytes(8)); // 16-character random password
-    
-        $hashedPassword = password_hash($password, PASSWORD_DEFAULT);
-       
-        $sql = "INSERT INTO voter (last_name, first_name, middle_name, email, password, role, status)
-                VALUES (?, ?, ?, ?, ?, ?, 'Active')";
-    
-        $stmt = $conn->prepare($sql);
-    
-        $stmt->bind_param("ssssss", $lastName, $firstName, $middleName, $email, $hashedPassword, $role);
-    
-        if ($stmt->execute()) {
-            // Set session variable to indicate account creation
-            $_SESSION['account_created'] = true;
-        
-            // Send email with password
-            $emailSender = new EmailSender($mail);
-            $emailSender->sendPasswordEmail($email, $password);
-        
-            // Redirect to admin-creation.php
-            header("Location: admin-creation.php");
-            exit;
+
+        // Email validation
+        $emailPattern = '/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}(?!\.c0m$)(?!@test)$/';
+        if (!preg_match($emailPattern, $email)) {
+            $emailError = 'Invalid email format.';
         } else {
-            // Handle error
+            // Check if email already exists
+            $stmt = $conn->prepare("SELECT COUNT(*) FROM voter WHERE email = ?");
+            $stmt->bind_param("s", $email);
+            $stmt->execute();
+            $stmt->bind_result($count);
+            $stmt->fetch();
+            $stmt->close();
+
+            if ($count > 0) {
+                $emailError = 'Email already exists in the database.';
```

Comment:
Refrain from mentioning anything technical such as the "database." Rephrase this next time.